### DISCUSSION TITLE
Date Time as an attribute type

### DISCUSSION THREAD

**Discussion by gewenyu99** [appwrite/appwrite#2944] [General]
Hi there~

I've been hearing people bring up the need for a DateTime attribute here and there. Both on Discord and on Reddit.

From what I understand, the current suggested approach is to use Unix timestamps. This is for performance and timezone independence.

I wanted to ask if you guys are running into issues with just UNIX timestamps, or if there's reasons you would like to see DateTime as a distinct data type in Appwrite.

**Comment by maeddin** [comment]
So I didn't ask for it and I don't need it, but I think an extra data type for it would be handy since it's easier to deal with. Even if it uses UNIX timestamps in the background, you could choose a user friendly notation in the console for example. Currently you can of course store it readable with ISO-8601, but this is not as efficient for querying. Almost all databases have a native DateTime-Type, which could ideally be used in the background.

But as I said: I personally don't need that. This is just an estimation of me, why some want to have it.

**Reply by gewenyu99** [reply]
Yep, this is the kind of feedback we're looking for. The underlying MariaDB instance supports DateTime, anyway. We're trying to judge if there's a good enough incentive to expose it. cc @Meldiron

**Reply by maeddin** [reply]
So I think that there are types that are more important and that you can't trivially store with an integer. For example, geographic coordinates come to mind, which currently have to be stored manually with a geohash in order to be able to search for position and radius. Of course there are other ways to store positions, e.g. lon and lat in an array or separately, so that you can at least search for all within a square and filter out the unnecessary ones.

**Reply by maeddin** [reply]
This was just an example for an attribute that might make more sense to implement instead of DateTime, which you can already use without problems.

**Reply by gewenyu99** [reply]
I've seen suggestions for Geo as well, I believe it is planned, too.

**Reply by 99hats** [reply]
I always end up needing DateTime for content. Usually blog and article types, but also chats, last seen and login stamps. It would be useful to have an automated timestamp attribute. The date and time of creation as determined by the server not the sender (which can potentially be spoofed).

It first thing I noticed about the attributes list was the lack of DateTime.

**Reply by CordMemescape** [reply]
I would definitely second the introduction of a DateTime type for Appwrite. It's a supported field/attribute in just about every database and the additional step of converting to and from a Unix Epoch seems unnecessary. Having a DateTime attribute would standardise the use of the dates and also make code clearer for queries.

**Reply by SMARTMICROWORLD** [reply]
No need of automated date time because it already has $createdAt and $updatedAt attribute bros and sis. Now feeling satisfied.😄

**Comment by gewenyu99** [comment]
This has been added, btw. Closing.
